src/travel-details-screens/components/Trip.tsx

```diff
@@ -51,6 +56,14 @@ export const Trip: React.FC<TripProps> = ({
     : undefined;
   const {vehiclePositions} = useGetServiceJourneyVehicles(ids);
 
+  const tripPatternLegs = tripPattern?.legs.map((leg) => {
+    let mode: AnyMode = !!leg.bookingArrangements ? 'flex' : leg.mode;
+    return {
+      ...leg,
+      mode,
+    };
+  });
+
```

Comment:
Now there are two places in the code doing this same mapping. Maybe take a look if it can either be done in a util-function, or maybe it is possible to do it earlier in the data fetching flow so it affects both the departure details and trip details screen.

Reply:
As next part of this task the `PaginatedDetailsHeader` will be removed from `TripDetailsScreenComponent`. That will include removing the duplication, so will leave it for now ❇️